### PR TITLE
[Up and running] Force user to complete profile upon first login

### PR DESCRIPTION
**Pull request by isc-rsingh**
This turned out to be a bit complicated. Since user profile data comes in via sync, you can't count on it being there at login. So I had to do a remote db query to get user profile data, and check it for required fields...

### CODE REVIEW

File: public/js/sodashboard.js [hunk 1]
```diff
@@ -353,23 +359,33 @@ db.get('_local/user').then(function(data) {
...
-      app.myTickets();
-    } else if (hash.match(/^edit/)) {
-      var match = hash.match(/[0-9]+$/);
-      if (match) {
-        app.edit(match[0]);
+  // if user has an incomplete profile, take them to their profile page
+  // get profile from remote db because sync is probably not yet complete
+  var rdb = new PouchDB(url);
+  rdb.get(app.loggedinuser["_id"]).then(function (userdata) {
+    if (!userdata.user_id || !userdata.user_name || !userdata.so_id) {
```
Comment (vabarbosa) [inline public/js/sodashboard.js]: should `user_id` and `user_name` be checked? they are obtained automatically from slack login and are **read only** (user has no ability to edit them- even if they are blank)

Comment (isc-rsingh) [reply public/js/sodashboard.js]: Good point. I'll remove those checks.

**Comment by ptitzler** [comment]
Bluemix [toolchain](https://console.bluemix.net/devops/toolchains/afc93986-3917-48ae-8901-f070f6d109f7 "green-blue-toolchain-20170516074606603"): [Delivery Pipeline](https://console.bluemix.net/devops/pipelines/5ea1a679-3a40-49d5-9621-b48806ab22cc "green-blue-toolchain-20170516074606603") deployed [sodashboard](https://github.com/ibm-cds-labs/sodashboard/releases/tag/deploy-ibm-yp-us-south-ptitzler_org-dev-20170605-205224 "deploy-ibm-yp-us-south-ptitzler_org-dev-20170605-205224") to [dev](https://console.ng.bluemix.net/apps/9c0f65df-3399-4895-87a9-7ba13bc0c226?env_id=ibm:yp:us-south "ibm:yp:us-south:ptitzler_org:dev"), including commit 6e1e48db5a32c44cab4092c0611e5a61931f5591

**Comment by ptitzler** [comment]
Bluemix [toolchain](https://console.bluemix.net/devops/toolchains/bb9178ab-039a-4b1b-bdfa-c5ab8e217c77 "sodashboard-wdp-infra"): [Delivery Pipeline](https://console.bluemix.net/devops/pipelines/592bddce-9340-4b45-9afa-14c722dd7005 "sodashboard-wdp-infra") deployed [sodashboard](https://github.com/ibm-cds-labs/sodashboard/releases/tag/deploy-ibm-yp-us-south-Dev-Advocates-wdp-infrastructure-20170605-205232 "deploy-ibm-yp-us-south-Dev-Advocates-wdp-infrastructure-20170605-205232") to [wdp-infrastructure](https://console.ng.bluemix.net/apps/9d6b467b-d433-4772-901c-a5bca087d5ae?env_id=ibm:yp:us-south "ibm:yp:us-south:Dev-Advocates:wdp-infrastructure"), including commit 6e1e48db5a32c44cab4092c0611e5a61931f5591

**Pull request merged**